galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistUtils.java

```diff
@@ -0,0 +1,288 @@
+				
+				JsonArray runIds = (JsonArray) jsonObject.get("runIds");		
+
+				for (JsonElement el : runIds) {
+					runIdsInWorklist.add(el.getAsString());
+				}	
+			}	
+			
+			return runIdsInWorklist;
+			
+		} else {
+			return null;
+		}
+		
+	}
+	
+	
+	protected List<WorklistItem> getWorklistItems(String principalUsername, List<String> runIdsInWorklist) throws ServletException {
+		
+		List<WorklistItem> runs = new ArrayList<WorklistItem>();
+		
+		List<String> runIdsNotFound = new ArrayList<String>();
+		
+		if (runIdsInWorklist != null) {
+			
+			IRunResult run = null;
+			for (String runId : runIdsInWorklist) {
+				try {
+					run = getRunById(runId);
+				} catch (ResultArchiveStoreException e) {
+					throw new ServletException("Error while retrieving run for Run ID " + runId, e);
+				}
+				
+				if (run != null) {
+					
+					try {
+						
+						String runName = run.getTestStructure().getRunName();
+						String shortName = run.getTestStructure().getTestShortName();
+						String testClass = run.getTestStructure().getTestName();
+						String result = run.getTestStructure().getResult();
+						
+						WorklistItem worklistItem = new WorklistItem(runId, runName, shortName, testClass, result);
+						
+						runs.add(worklistItem);
+						
+					} catch (ResultArchiveStoreException e) {
+						throw new ServletException("Error while retriving Worklist information for Run ID " + runId, e);
+					}
+					
+				} else {
+					runIdsNotFound.add(runId);
+					try {
+						removeRunIdFromDss(principalUsername, runId);
+					} catch (Exception e) {
+						throw new ServletException("Error while updating the DSS", e);
+					}
+				}
+				
+			}
+			
+			runIdsInWorklist.removeAll(runIdsNotFound);
+		}
+		
+		return runs;
+	}
+	
+	protected IRunResult getRunById(String id) throws ResultArchiveStoreException {
+
+		IRunResult run = null;
+	       
+        for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
+           
+           run = directoryService.getRunById(id);
+           
+           if(run != null) {
+        	   return run;
+           }
+           
+        }
+		
+		return null;
+	}
+	
+	protected void removeRunIdFromDss(String principalUsername, String runId) throws DynamicStatusStoreException, ServletException {
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		String oldValue = dss.get("user." + principalUsername + ".worklist");
+		
+		if (oldValue != null) {
+			
+			JsonObject jsonObject = new JsonParser().parse(oldValue).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			JsonArray runIdsToRemove = new JsonArray();
+			for (JsonElement el : runIds) {
+				if (el.getAsString().equals(runId)) {
+					runIdsToRemove.add(el);
+				}
+			}
+			
+			if (runIdsToRemove.size()>0) {
+				for (JsonElement el : runIdsToRemove) {
+					runIds.remove(el);
+				}
+			}
+		
```

Comment:
if you don't find the runid in the list,  there is no point updating the dss with the same value,  so you should return out